superset/utils/decorators.py

```diff
@@ -61,6 +65,82 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
     return decorate
 
 
+def logs_context(
+    context_func: Callable[..., dict[Any, Any]] | None = None,
+    **ctx_kwargs: int | str | UUID | None,
+) -> Callable[..., Any]:
+    """
+    Takes arguments and adds them to the global logs_context.
+    This is for logging purposes only and values should not be relied on or mutated
+    """
+
+    def decorate(f: Callable[..., Any]) -> Callable[..., Any]:
+        def wrapped(*args: Any, **kwargs: Any) -> Any:
+            if not hasattr(g, "logs_context"):
+                g.logs_context = {}
+
+            # limit data that can be saved to logs_context
+            # in order to prevent antipatterns
+            available_logs_context_keys = [
```

Comment:
Would it make sense for this to be a parameter? Right now this is hard coded for query related logging, but it might be useful for other logging purposes. 

If we add the `available_logs_context_keys` as a parameter when the decorator is called we can still keep the functionality of limiting data but for that specific function.

Reply:
Yeah, that's a good question. In the SIP review we talked about trying to limit antipatterns of people putting random data into global context so the intention here is to use this list to limit what data can be passed in with the decorator, but there are definitely a lot of other use cases. I have three others on my todo list. I would suggest that if you think there's more data that should be considered for logging, then add it to the list in a PR and we'll see if it fits. That way we're still limiting the data that the decorator can accept, but we're not completely limited to just the set that is here now. How does that sound?

Reply:
Hey Elizabeth, that sounds great! As for now there's no other data we need aside from dataset_id/slice_id so this is good on our end. Was just pointing this out as a future suggestion.